app/controllers/application_controller.rb

```diff
@@ -28,7 +29,9 @@ def append_info_to_payload(payload)
   attr_writer :analytics
 
   def analytics
-    @analytics ||= Analytics.new(current_user, request)
+    user = current_user || AnonymousUser.new
```

Comment:
Now this, I think, is a good use of the `Active Nothing` pattern :)

I think maybe we should we change current_user to do this automatically?